### PR TITLE
remove openshift infra command

### PR DESCRIPTION
**Pull request by deads2k**
This removes the `openshift infra` command. It does not replace 
```
-		irouter.NewCommandTemplateRouter("router"),		
-		irouter.NewCommandF5Router("f5-router"),		
-		deployer.NewCommandDeployer("deploy"),		
-		recycle.NewCommandRecycle("recycle", out),		
-		builder.NewCommandS2IBuilder("sti-build"),		
-		builder.NewCommandDockerBuilder("docker-build"),
```
since those have their own binaries.

@sosiouxme I created a new command `openshift-diagnostics` to hold `openshift-diagnostics diagnostic-pod` and `openshift-diagnostics network-diagnostic-pod`.  I'm not attached to the name, but looking at the dependency tree, I'm betting these move with `oc` later this month.

@smarterclayton I think this fulfills the intent of our conversation last week.
@juanvallejo fyi

### CODE REVIEW

File: hack/lib/constants.sh [hunk 1]
```diff
@@ -43,6 +43,7 @@ readonly OS_CROSS_COMPILE_TARGETS=(
   cmd/openshift
   cmd/oc
   cmd/kubefed
+  cmd/openshift-diagnostics
```
Comment (smarterclayton) [inline hack/lib/constants.sh]: Hrm, confusing that this is just the binary used by diagnostics in pods.  It’s fine for now.

**Comment by smarterclayton** [comment]
Lgtm

File: hack/lib/constants.sh (hunk 1 above)
Comment (deads2k) [reply hack/lib/constants.sh]: > Hrm, confusing that this is just the binary used by diagnostics in pods. It’s fine for now.

Yeah.  I'm still open to different ways of structuring it as we push through 3.9.  For now, I need them out of openshift so I can snip `kubectl` dependencies and they likely belong together under a related suite of commands.

File: cmd/openshift-diagnostics/main.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/openshift/origin/pkg/cmd/util/serviceability"
+	"github.com/openshift/origin/pkg/oc/admin/diagnostics"
+	"github.com/spf13/cobra"
+
+	// install all APIs
+	_ "github.com/openshift/origin/pkg/api/install"
+	_ "k8s.io/kubernetes/pkg/api/install"
+	_ "k8s.io/kubernetes/pkg/apis/autoscaling/install"
+	_ "k8s.io/kubernetes/pkg/apis/batch/install"
+	_ "k8s.io/kubernetes/pkg/apis/extensions/install"
```
Comment (mfojtik) [inline cmd/openshift-diagnostics/main.go]: do we need apps ?

**Comment by mfojtik** [comment]
/lgtm

need update of complete

File: cmd/openshift-diagnostics/main.go (hunk 2 above)
Comment (deads2k) [reply cmd/openshift-diagnostics/main.go]: > cmd/openshift-diagnostics/main.go

No, the main one installs all.  We are in need of an overhaul for installation though.

**Comment by sosiouxme** [comment]
Cool, thanks for the heads up. Is there a card or doc summarizing the intent of the changes?

**Comment by deads2k** [comment]
> Cool, thanks for the heads up. Is there a card or doc summarizing the intent of the changes?

It's related to this: https://trello.com/c/xemjduCn/1068-21-move-oc-to-external-repository 

We need to split `oc` out  of `origin` and we're chasing down everything with a dependency from `openshift` on `oc` and this fell out.

File: hack/lib/constants.sh (hunk 1 above)
Comment (sosiouxme) [reply hack/lib/constants.sh]: @deads2k @smarterclayton got no problem with breaking this out, and don't much care about the name, but it needs to actually be included in the image which it isn't in 3.9-alpha.0... how can I make that happen?

Comment (deads2k) [reply hack/lib/constants.sh]: > @deads2k @smarterclayton got no problem with breaking this out, and don't much care about the name, but it needs to actually be included in the image which it isn't in 3.9-alpha.0... how can I make that happen?

I would try to work backwards from `images/origin/Dockerfile`.  We're installing the RPM, so if it isn't present, I suspect the RPM doesn't include it, which probably means fixing up origin.spec.

**Comment by sosiouxme** [comment]
On Fri, Dec 15, 2017 at 10:39 AM, David Eads <notifications@github.com>
wrote:

> I would try to work backwards from images/origin/Dockerfile. We're
> installing the RPM, so if it isn't present, I suspect the RPM doesn't
> include it, which probably means fixing up origin.spec.
>
>
Thanks, was looking at it yesterday and that was where I was headed. It's
a 162M binary; I was wondering if that was OK to add to the RPM (already
kind of huge) or if it should be folded back into an existing binary and
packaged with a symlink.

**Comment by deads2k** [comment]
> Thanks, was looking at it yesterday and that was where I was headed. It's
> a 162M binary; I was wondering if that was OK to add to the RPM (already
> kind of huge) or if it should be folded back into an existing binary and
> packaged with a symlink.

It was split out because the openshift binary cannot have any dependency on `k8s.io/kubernetes/pkg/kubectl`.  That means it cannot be included in the `openshift` command.  Also, having single binaries that do different things based on name seems highly suspect.

**Pull request merged**